message_ix_models/model/water/data/water_supply.py

```diff
@@ -97,7 +97,9 @@ def map_basin_region_wat(context: "Context") -> pd.DataFrame:
         if context.type_reg == "country":
             df_sw["MSGREG"] = context.map_ISO_c[context.regions]
         else:
-            df_sw["MSGREG"] = f"{context.regions}_" + df_sw["BCU_name"].str[-3:]
+            df_sw["MSGREG"] = (
```

Comment:
This line counts as not tested, but you're actually always setting `df_sw["MSGREG"]`, so it should count. You can make pytest recognize that by switching again to

```python
df_sw["MSGREG"] = context.map_ISO_c[context.regions] if context.type_reg == "country" else f"{context.regions}_" + df_sw["BCU_name"].str.split("|").str[-1]
```

(Can't leave suggestions on deleted lines, unfortunately. Please apply ruff here, too.)